### PR TITLE
feat: add `mode` method

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

I started to take a very mild look at skrub... and immediately stumbled upon not having `mode` implemented
  
## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/_arrow/series.py [hunk 1]
```diff
@@ -667,6 +673,13 @@ def clip(
     def to_arrow(self: Self) -> pa.Array:
         return self._native_series.combine_chunks()
 
+    def mode(self: Self) -> ArrowSeries:
+        plx = self.__narwhals_namespace__()
+        col_token = generate_unique_token(n_bytes=8, columns=[self.name])
+        return self.value_counts(name=col_token, normalize=False).filter(
+            plx.col(col_token) == plx.col(col_token).max()
+        )[self.name]
```
Comment (FBruzzesi) [inline narwhals/_arrow/series.py:681]: I was not able to make it work using [`pyarrow.compute.mode`](https://arrow.apache.org/docs/python/generated/pyarrow.compute.mode.html) directly as it requires to know how many distinct most-common values to return.

File: narwhals/_dask/expr.py [hunk 2]
```diff
@@ -644,6 +644,10 @@ def func(df: DaskLazyFrame) -> list[Any]:
             backend_version=self._backend_version,
         )
 
+    def mode(self: Self) -> Self:
+        msg = "`Expr.mode` is not supported for the Dask backend."
+        raise NotImplementedError(msg)
```
Comment (FBruzzesi) [inline narwhals/_dask/expr.py:649]: Of course, mode changes the index (as it changed the length)

**Review by MarcoGorelli — APPROVED** [review]
thanks @FBruzzesi !

**Comment by FBruzzesi** [comment]
Crazy timing! Polars just changed behaviour on mode with multiple columns.

From the docs itself:

```py
df = pl.DataFrame(
    {
        "a": [1, 1, 2, 3],
        "b": [1, 1, 2, 2],
    }
)
df.select(pl.all().mode())
```

- For polars before 1.7, result is as in the test we have
- For polars 1.7 it raises   `InvalidOperationError: Series: a, length 1 doesn't match the DataFrame height of 2`. 

Should we revert this PR?

**Comment by MarcoGorelli** [comment]
i think we can just remove the multi-column test - not sure if it's a bug in Polars, will check

**Pull request merged**